lldb/source/Plugins/Language/Swift/SwiftRuntimeFailureRecognizer.cpp

```diff
@@ -0,0 +1,93 @@
+#include "SwiftRuntimeFailureRecognizer.h"
+
+#include "lldb/Core/Module.h"
+#include "lldb/Symbol/Function.h"
+#include "lldb/Symbol/SymbolContext.h"
+#include "lldb/Target/Process.h"
+#include "lldb/Target/Target.h"
+#include "lldb/Target/Thread.h"
+
+#include "lldb/Utility/Log.h"
+#include "lldb/Utility/Logging.h"
+
+using namespace llvm;
+using namespace lldb;
+using namespace lldb_private;
+
+SwiftRuntimeFailureRecognizedStackFrame::
+    SwiftRuntimeFailureRecognizedStackFrame(StackFrameSP most_relevant_frame_sp,
+                                            StringRef stop_desc)
+    : m_most_relevant_frame(most_relevant_frame_sp) {
+  m_stop_desc = stop_desc;
+}
+
+lldb::RecognizedStackFrameSP SwiftRuntimeFailureFrameRecognizer::RecognizeFrame(
+    lldb::StackFrameSP frame_sp) {
+  if (frame_sp->GetFrameIndex())
+    return RecognizedStackFrameSP();
+
+  ThreadSP thread_sp = frame_sp->GetThread();
+  ProcessSP process_sp = thread_sp->GetProcess();
+
+  StackFrameSP most_relevant_frame_sp = thread_sp->GetStackFrameAtIndex(1);
+
+  if (!most_relevant_frame_sp) {
+    Log *log(lldb_private::GetLogIfAllCategoriesSet(LIBLLDB_LOG_UNWIND));
+    LLDB_LOG(
+        log,
+        "Swift Runtime Failure Recognizer: Hit unwinding bound (1 frame)!");
+    return RecognizedStackFrameSP();
+  }
+
+  SymbolContext sc = frame_sp->GetSymbolContext(eSymbolContextEverything);
+
+  if (!sc.block)
+    return RecognizedStackFrameSP();
+
+  const InlineFunctionInfo *inline_info = nullptr;
```

Comment:
Can you add a comment here explaining that the error is encoded as an inline frame and perhaps even reference the place in the compiler where it is generated so they can be kept in sync?

Reply:
What I meant was either a filename like "swift/lib/IRGen/IRGenSIL.cpp" or something along the lines of "see the XYZ class in the swift compiler's SILGen library". I don't know how permanent GitHub pull request links are relative to the lifetime of the Swift project.